docs: rewrite common-misconceptions.md to ASD-STE100 rules (#2035)
apache/maven-resolver: src/site/markdown/common-misconceptions.md

During the "collect" step, various criteria select the nodes. The configured scope filters are among these criteria. This leads to the notion of the "runtime graph" and the "test graph".

The "scope filter" selects what to omit. This use of the filter is not intuitive. Maven Core uses the filter this way. The default session filter in Maven is set up as follows:

```
  new ScopeDependencySelector("test", "provided")
```

This filter omits the dependencies of the "current dependency node" that are in the "test" and "provided" scope. In other words, this filter builds the "downstream runtime classpath" of the supplied artifact. It shows what the artifact needs at runtime when you depend on it.

Note: These notions do not relate to Maven yet. Maven does not appear in this example. This is not the classpath that the Compiler or Surefire plugins use. It is only a showcase of how Resolver works.

## Misconception Number 2: "Test graph" is a Superset of the "Runtime graph"

> **Wrong**. For the runtime graph, Resolver omits the "test" scoped dependencies. This has interesting consequences. The example below shows this.

Reviewer: The original stated: _"It was true in Maven2, where test graph really was a superset of runtime, but this does not stand anymore in Maven3."_

This historical context is important on a page titled "Common Misconceptions" — it explains *why* people hold this belief (because it was actually true in Maven 2). Without it, the section says "Wrong" but doesn't explain the origin. Consider restoring it:

```suggestion
**Wrong**. In Maven 2, the test graph was a superset of the runtime graph. This is no longer true in Maven 3. For the runtime graph, Resolver omits the "test" scoped dependencies. This has interesting consequences. The example below shows this.
```

      </dependency>
```

> The model code does not directly use Guava. However, Guava is a direct dependency of Guice. 

Reviewer: "The model code" is not standard Maven/Java terminology. In the Maven ecosystem, "model" has a specific meaning (the POM model, `org.apache.maven.model.Model`), so "model code" could be misread as referring to the POM model code.

The original said "you want to avoid any use of Guava" — referring to the project's production source code. Consider:

```suggestion
Your project code does not directly use Guava. However, Guava is a direct dependency of Guice.
```

> The model code does not directly use Guava. However, Guava is a direct dependency of Guice. 

Reviewer: Nit: trailing whitespace at the end of this line (after "Guice.").
